### DISCUSSION TITLE
Request Logging, v1

### DISCUSSION THREAD

**Discussion by DGuhr** [keycloak/keycloak#10473] [Archive]
A very common scenario is to troubleshoot the requests from and to Keycloak, a.k.a. a HTTP Access log. Quarkus gives us some things out of the box, see https://quarkus.io/guides/http-reference#configuring-http-access-logs

Proposal for V1:

Have a new configuration property `log-request-enabled=true/false` that internaly maps to `quarkus.http.access-log.enabled` and a dedicated log category set called `org.keycloak.request` (by using `quarkus.http.access-log.category=org.keycloak.request` internally.

There are other options to consider, but this is a basic V1, enabling ppl to just use `kc.sh start --log-requests-enabled=true`. wdyt?

**Comment by stianst** [comment]
We have a number of request "categories" such as:

* Incoming http requests
* Outgoing http requests
* LDAP requests
* DB queries
* Infinispan?

I would imagine that commonly people will want to enable one of these, and not all.

Not sure if we would want something like `--log-request-categories=http,http-client`, or something like `--http-access-log-enabled=true --http-client-access-log-enabled=true`.

**Reply by stianst** [reply]
With regard to the log category if we use `org.keycloak.request` for incoming http requests what do we do with others? In general I think it will be quite hard to map all categories to a KC log category, but I can see the benefit of doing so.

**Comment by kkcmadhu** [comment]
until this is developed, whats  the best way to have acess logs enabled? 
do you suggest having a quarkus conf file and enable acccess-log natively on quarkus?
